### PR TITLE
Add Pr check for agent-base

### PR DESCRIPTION
**Pull request by DmitriiBobreshev**
**Task name**: -

**Description**: 
- added pr checks for agent-base
- added troubleshooting section to the docs

**Documentation changes required:** N

**Added unit tests:** N

**Attached related issue:** [AB#2198533](https://dev.azure.com/mseng/b924d696-3eae-4116-8443-9a18392d8544/_workitems/edit/2198533)

**Checklist**:
- [ ] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

File: ci/after-build-check-tasks.js [hunk 1]
```diff
@@ -105,33 +126,74 @@ function findNonUniqueTaskLib() {
...
+        logToPipeline('error', 'Please check the tasks, seems like they have invalid PowerShell syntax.');
         process.exit(1);
     }
 }
 
-// findNonUniqueTaskLib();
-analyzePowershellTasks();
\ No newline at end of file
+function findIncompatibleAgentBase() {
+    const minAgentBaseVersion = '6.0.2';
```
Comment (KonstantinTyukalov) [inline ci/after-build-check-tasks.js:168]: is this version compatible with node 10-16?

File: ci/after-build-check-tasks.js [hunk 2]
```diff
@@ -105,33 +126,74 @@ function findNonUniqueTaskLib() {
...
+        logToPipeline('info', 'The powershell check is only supported on Windows. Skipping...');
         return;
     }
 
     try {
         const pwshScriptPath = path.join(__dirname, 'check-powershell-syntax.ps1');
         output = util.run(`powershell -NoLogo -Sta -NoProfile -NonInteractive -ExecutionPolicy Unrestricted ${pwshScriptPath} ${buildTasksPath}`, true);
     } catch (e) {
-        console.log(`##vso[task.logissue type=error;sourcepath=ci/check-tasks.js;linenumber=123;]Please check the tasks, seems like they have invalid PowerShell syntax.`)
+        logToPipeline('error', 'Please check the tasks, seems like they have invalid PowerShell syntax.');
```
Comment (KonstantinTyukalov) [inline ci/after-build-check-tasks.js:162]: [not blocking] should we show output error from tool for better understanding?

**Review by KonstantinTyukalov — APPROVED** [review]

File: ci/after-build-check-tasks.js (hunk 1 above)
Comment (DmitriiBobreshev) [reply ci/after-build-check-tasks.js:168]: yes

File: ci/after-build-check-tasks.js (hunk 2 above)
Comment (DmitriiBobreshev) [reply ci/after-build-check-tasks.js:162]: the util.run call with [inheritStreams:true](https://github.com/microsoft/azure-pipelines-tasks/blob/master/ci/ci-util.js#L103) will show all output from PowerShell command, including the error, so we don't need to additionally output error

File: ci/after-build-check-tasks.js [hunk 3]
```diff
@@ -1,11 +1,12 @@
 const util = require('./ci-util');
 const fs = require('fs');
 const path = require('path');
-var crypto = require('crypto');
+const semver = require('semver');
```
Comment (DergachevE) [inline ci/after-build-check-tasks.js:4]: What is the version of `semver` package is used?

**Review by DergachevE — APPROVED** [review]

File: ci/after-build-check-tasks.js (hunk 3 above)
Comment (DmitriiBobreshev) [reply ci/after-build-check-tasks.js:4]: The ci scripts share semver package with the main build system, so [it's 4.3.3](https://github.com/microsoft/azure-pipelines-tasks/blob/master/package.json#L45)

**Pull request merged**